### PR TITLE
fix AQL leak

### PR DESCRIPTION
**Pull request by jsteemann**
Fix leaking AqlItemBlocks in SortingGatherBlock.

### CODE REVIEW

File: arangod/Aql/ClusterBlocks.cpp [hunk 1]
```diff
@@ -1376,6 +1376,21 @@ SortingGatherBlock::SortingGatherBlock(
...
+  clearBuffers();
+}
+
+void SortingGatherBlock::clearBuffers() noexcept {
+  for (std::deque<AqlItemBlock*>& x : _gatherBlockBuffer) {
+    for (AqlItemBlock* y : x) {
+      delete y;
+    }
+    x.clear();
+  }
```
Comment (goedderz) [inline arangod/Aql/ClusterBlocks.cpp]: Wouldn't it be simpler and safer to change the type of `_gatherBlockBuffer` from `std::vector<std::deque<AqlItemBlock*>>` to `std::vector<std::deque<std::unique_ptr<AqlItemBlock>>>` and let the default destructors handle it?

Comment (mchacki) [reply arangod/Aql/ClusterBlocks.cpp]: I agree.
Should be save to use, it is handed over by reference to the Sorting implementations.

**Comment by jsteemann** [comment]
@mchacki @goedderz I thought about changing `_gatherBlockBuffer` to an `std::vector<std::deque<std::unique_ptr<AqlItemBlock>>>` too, but that required many changes, and adjustments to users of the gatherBlockBuffer that use `emplace_back`, `reserve` etc.
So I didn't want to do this bigger change now, and simply fix the leak.

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/294/

**Pull request merged**